contracts/utils/structs/EnumerableSet.sol

```diff
@@ -63,6 +63,26 @@ library EnumerableSet {
         }
     }
 
+    /**
+     * @dev Removes all elements from the set. O(n).
+     *
+     * Note that this can run out of gas for large sets.
+     * First use `remove` to reduce the size of a large set in this case.
+     */
+    function _clear(Set storage set) private {
+        uint256 last = set._values.length;
+        unchecked {
+            while (last --> 0) {
```

Comment:
tbh, I've spent 15 minutes figuring out what is this `-->` syntax. Eventually my colleague saved me as he guessed it is `last-- > 0`. I wasn't even aware that it is possible. While this language feature is cool and I really like the way it looks, it seems to be too advanced for the most popular contract library.

Reply:
This is the best way to loop to zero inclusive on an unsigned variable. I've always thought it more clear when written `-->`, but I would rearrange the whitespace upon reviewer request.

Reply:
Additional matter of interest: decrement inside solidity loop conditions doesn't seem to work at all in 0.8.0. I am concerned at how the tests still passed.